### ISSUE TITLE
[FEATURE] remove ujson dependency (it is dead)

### ISSUE THREAD

**Issue by hnykda** [tiangolo/fastapi#820]
FastAPI is using `ujson` which has been not actively developed for about 3 years by now. [Here is a relevant issue](https://github.com/esnme/ultrajson/issues/343)

**Comment by dmontagu** [comment]
I would be in favor of recommending `orjson` instead of `ujson`.

I recall the creator of `orjson` reaching about this many months ago and there was some confusion due to slow responses, and also some issues on the pydantic side, but I think many/most of the barriers to usage have since been removed. If `orjson` seems to be taking over as the go-to high-performance json library, I think it would make sense to use that for development and recommend it.

To the extent that there is any difference in json encoding/decoding between json and orjson (or ujson and orjson), it would probably be best to document it.

I'd be willing to review a PR making this change. (I'd leave the ultimate decision about merging to @tiangolo though.)

**Comment by tiangolo** [comment]
Yeah, I've been intending to do it for quite some time now... :sweat_smile: 

In fact, you can sort of do it right now (and since several versions ago). 

---

Here's my plan:

I want to export everything from Starlette, to avoid confusions of the type "do I import this from FastAPI or from Starlette?", I've seen reports of that and even I've struggled with that. For example, FastAPI has its own `HTTPException`.

With that, I want to have a response `ORJSONResponse` exported from `fastapi.responses`, among Starlette's responses.

In fact, I added tests for creating custom responses (as part of a PR), testing specifically creating orjson responses: https://github.com/tiangolo/fastapi/blob/f803c77515662cd2382674d9fc8df1e6641b3ba7/tests/test_default_response_class.py#L9-L13
And you can replicate creating that response class and using it. It's just 3 lines (you would remove the custom `media_type`), but still, it should be provided by FastAPI (among all the other responses from Starlette).

```Python
app = FastAPI(default_response_class=ORJSONResponse)
```

---

Also, as a side note, UJSON is not really used by default anywhere. It's optional, and you would have to use the `UJSONResponse` class for it to be used, the same way you would use the (new/custom/future) `ORJSONResponse`.

**Comment by hnykda** [comment]
Well, that's elegant :clap: .

> Also, as a side note, UJSON is not really used by default anywhere. It's optional, and you would have to use the UJSONResponse class for it to be used, the same way you would use the (new/custom/future) ORJSONResponse.
Then I don't understand why it's in [requirements](https://github.com/tiangolo/fastapi/blob/master/pyproject.toml#L81).

**Comment by dmontagu** [comment]
@hnykda the “all” set of requirements there includes everything necessary to run *all* tests, including of features that are optional. It is intended more for development purposes than for use by consumers.

**Comment by hnykda** [comment]
Ah, got it! Thanks. I believe this can be closed then.

Personally, I would not reference any obsolete dependencies in my project even if they are optional (especially so "future-looking" like Python 3.6+, latest libs, ...) . But that's surely subjective.

**Comment by dagnelies** [comment]
Apparently, it found a new maintainer and is now alive and kicking again. Well, dunno about the kicking part, but releases are pretty fresh: 

https://pypi.org/project/ujson/

and repo seems active again.

**Comment by dagnelies** [comment]
Nevermind, apparently `orjson` is a fine choice too. Apparently one of the fastest.

**Comment by ahmadazizi** [comment]
Just to mention, installing fastapi on windows platform (python 3.9) fails, since orjson neither provide the compatible wheel nor could be compiled.

**Comment by ahtik** [comment]
@ahmadazizi,  `orjson` wheels are now available also for Windows Python 3.9.

**Comment by ycd** [comment]
Since Starlette dropped support for `UJSONResponse` in https://github.com/encode/starlette/pull/1047 (Requires Starlette 0.14.1), we can remove it too, `ujson` causes build fails on many distributions due C Dependencies (See #2811 ). 

This require changes in #2335, since it re-adds `UJSONResponse`.

**Comment by omidmaldar** [comment]
Regarding ujson there is a [security alert](https://github.com/advisories/GHSA-fh56-85cw-5pq6) with moderate severity which is patched in 5.1.0.  Fastapi 0.74.0 requires `ujson (>=4.0.1,<5.0.0)` which is not safe.

**Comment by melt7777** [comment]
Hello all, I did find a workaround which allows us to build images without the ujson. I am using the `python-poetry` package manager, instead of requirements.txt. 

FastAPI is fantastic so far, thank you all for your hard work on this project. 

Here, roughly the steps I used:

1. `poetry init` 
2. `poetry add fastapi -E all` - it installed the ujson package and placed the ujson bits into the pyproject.toml and the newly created poetry.lock file. 
3. Manually edit the .toml and .lock files and remove all the ujson items.
4. Build image as usual and it will not have the ujson or its vuln in the image. 

Hope this helps someone until they scan make the permanent fix.

Thank you,

~ melt
